src/netif.c

```diff
@@ -1403,6 +1434,79 @@ static int set_ipaddr_ioctl(const struct ip_setting_handler *handler, struct net
     return 0;
 }
 
+static int remove_ipaddr(const struct ip_setting_handler *handler, struct netif *nb, const char *ifname, void *context)
+{
+  const char *ipaddr = (const char *) context;
+  char *prefix = (const char *) strchr(context, ':');
+
+  struct ifreq ifr;
+  memset(&ifr, 0, sizeof(ifr));
+  strncpy(ifr.ifr_name, ifname, IFNAMSIZ);
+
+  struct sockaddr_in *addr = (struct sockaddr_in *) &ifr.ifr_addr;
+  struct nlmsghdr    *nlh  = mnl_nlmsg_put_header(nb->nlbuf);
+  struct ifaddrmsg   *ipm  = mnl_nlmsg_put_extra_header(nlh, sizeof(struct ifaddrmsg));;
+
+  int seq = nb->seq++;
+  int ret = 0;
+
+  nlh->nlmsg_type  = RTM_DELADDR;
+  nlh->nlmsg_flags = NLM_F_REQUEST;// | NLM_F_ACK;
```

Comment:
any reason to keep the commented out flag?

Reply:
fair point :) Re-adding the ACK ;)